src/java.base/windows/classes/sun/nio/ch/WindowsAsynchronousFileChannelImpl.java

```diff
@@ -662,6 +668,7 @@ public void run() {
                     ioCache.remove(overlapped);
 
             } finally {
+                NIO_ACCESS.releaseSession(buf);
```

Comment:
I thin the change to WindowsAsynchronousFileChannel will need more than one reviewer as the I/O operation may complete on a different thread to the one that it was initiated on.